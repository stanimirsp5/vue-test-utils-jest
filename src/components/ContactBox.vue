<template>
    <div :class="{ selected: selected }"
        @click="handleClick">
        <p>{{ fullName }}</p>
    </div>
</template>
<script>
export default {
    props: ["id", "name", "surname", "selected"],
    computed: {
        fullName() {
            return `${this.name} ${this.surname}`;
        }
    },
    methods: {
        handleClick() {
            this.$emit("contact-click", this.id);
        }
    }
};
</script>
