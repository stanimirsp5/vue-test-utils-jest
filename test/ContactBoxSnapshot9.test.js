import {
  mount
} from "@vue/test-utils";
import ContactBox from "@/components/ContactBox";

const createContactBox = (id, name, surname, selected) => mount(ContactBox, {
  propsData: {
    id,
    name,
    surname,
    selected
  }
});
// snapshottestingisusefultocheckchangesonthe rendering state.
// If the rendering state doesn’t change,
// there is no way that snapshot testing can help us.
// Why snapshot testing?
// most of the times if you change the code you must
//change the assertions on the tests, while with snapshot testing you don’t need to.
describe("ContactBox.test.js", () => {


  describe("Watchers - inputValue", () => {
    it("fullName should be the combination of name + surname", () => {
      const cmp = createContactBox(0, "John", "Doe", false);
      //expect(cmp.vm.fullName).toBe("John Doe");
      expect(cmp.element).toMatchSnapshot();
    });

    it("should have a selected class when the selected prop is true", () => {
      const cmp = createContactBox(0, "John", "Doe", true);
      // expect(cmp.classes()).toContain("selected");
      expect(cmp.element).toMatchSnapshot();
    });

    it("should emit a contact-click event with its id when the component is cli\ cked", () => {
      const cmp = createContactBox(0, "John", "Doe", false);
      cmp.trigger("click");
      const payload = cmp.emitted("contact-click")[0][0];
      expect(payload).toBe(0);
    });
  });



});
