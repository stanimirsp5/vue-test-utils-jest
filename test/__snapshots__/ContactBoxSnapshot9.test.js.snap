// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactBox.test.js Watchers - inputValue fullName should be the combination of name + surname 1`] = `
<div
  class=""
>
  <p>
    John Doe
  </p>
</div>
`;

exports[`ContactBox.test.js Watchers - inputValue should have a selected class when the selected prop is true 1`] = `
<div
  class="selected"
>
  <p>
    John Doe
  </p>
</div>
`;
